### ISSUE TITLE
USB-C Ethernet Adapter Not Detected On Lenovo X1C6

### ISSUE THREAD

**Issue by manasb** [QubesOS/qubes-issues#4426]
### Qubes OS version:

`R4.0`

### Affected component(s): 

I am trying to use a [Lenovo USB-C to Ethernet Adapter][adapter] on a Lenovo X1C6 running Qubes `R4.0` with `sys-usb` enabled.

[adapter]: https://www.lenovo.com/us/en/accessories-and-monitors/cables-and-adapters/adapters/CABLE-BO-USB-C-to-Ethernet/p/4X90S91831

---

### Steps to reproduce the behavior:

- Attach USB-C to Ethernet adapter to system

### Expected behavior:

- Device is detected by the system and is displayed as a new device in `sys-usb`

### Actual behavior:

- Device is not detected at all. I have tested with an ethernet cable plugged in to the adapter but the device is still not detected by the system and is not displayed in `sys-usb`.

### General notes:

I have not ruled out a faulty adapter, this is brand new so I would expect it to be functional. I do not have any other system which has a USB-C port nor do I have any other USB-C devices so I am limited as far as being able to test different configurations -- the supplied charger for this laptop has a USB-C connector which has been working fine.

---

### Related issues:

None at the moment.

---

P.S. I've seen some unfriendly behavior towards me in this repository from other users whom I have never interacted with before. I just wanted to say thank you to the Qubes team, especially @andrewdavidwong, for your exemplary level of professionalism :+1:

**Comment by marmarek** [comment]
I suspect it isn't USB device, but PCIe one (thunderbolt also have such interface). And kernel used in Qubes have PCI(e) hotplug disabled: #1673
If that's indeed the case, just enabling PCI hotplug may not be enough - you'd need to assign new devices to sys-usb. And ideally this should be done automatically, without attaching device to dom0 first (not sure if that is possible...).

**Comment by andrewdavidwong** [comment]
> I suspect it isn't USB device, but PCIe one (thunderbolt also have such interface). And kernel used in Qubes have PCI(e) hotplug disabled: #1673
> If that's indeed the case, just enabling PCI hotplug may not be enough - you'd need to assign new devices to sys-usb. And ideally this should be done automatically, without attaching device to dom0 first (not sure if that is possible...).

Does that mean this issue is a `wontfix`?

**Comment by marmarek** [comment]
> Does that mean this issue is a `wontfix`?

I'd really like it not to be the case, but that require some research and probably won't happen soon.
Rough plan how this could be done:
 - check what needs to be done to passthrough PCIe bus/root port instead of single device, including appropriate IOMMU setting (if that's possible)
 - research linux and xen support for thunderbolt security levels, and how this relate to PCI passthrough; for example if it's possible to enable the device only after connecting it to sys-usb
 - for both of the above - extend linux kernel to control PCI hotplug separately for different kind of devices (thunderbolt, express card, docking station, etc)

**Comment by andrewdavidwong** [comment]
Ok, leaving it open on the "Far in the future" milestone.

-----

> I just wanted to say thank you to the Qubes team, especially @andrewdavidwong, for your exemplary level of professionalism :+1:

Thanks! :slightly_smiling_face:

**Comment by manasb** [comment]
I have tried using the ethernet port of this dock: https://www.lenovo.com/us/en/accessories-and-monitors/top-tech/Pro-Dock-135-W-US/p/40AH0135US and that works fine automatically. I'll look into using a non type-c USB ethernet adapter for now when I'm not using the docking station. Thanks

**Comment by noseshimself** [comment]
I didn't find this due to its heading ("assign new PCIe devices to sys-usb"). The proposal does not make much sense unless there is an additional sys-pci (and probably more of them for different bus systems).

Besides that: USB attachment is depending on the hardware design. Take Lenovo Thinkpads as an example: While on the P70 attaching a USB device is exposing a new PCI device with USB connected to it plugging a _**pure**_ USB device (e. g. a NIC that will work with a C-to-A adapter) into a C connector will show up in sys-usb immediately.

An easy solution might be adding the option of attaching PCI devices to VMs which are not available on boot and hotplugging them later...

**Comment by brendanhoar** [comment]
> While on the P70 attaching a USB device is exposing a new PCI device with USB connected to it plugging a _**pure**_ USB device (e. g. a NIC that will work with a C-to-A adapter) into a C connector will show up in sys-usb immediately.

I am having trouble parsing the example sentence above. Can you elaborate?

**Comment by 3hhh** [comment]
Related discussion: https://groups.google.com/forum/#!topic/qubes-users/19TisWm-_fs

From my point of view this is an instance of #2454 though.
